src/java/org/streamingpool/core/domain/StreamDependencyTree.java

```diff
@@ -0,0 +1,17 @@
+package org.streamingpool.core.domain;
+
+import org.streamingpool.core.service.StreamId;
+
+import java.util.Set;
+
+/**
+ * Data structure that holds the dependencies between {@link StreamId}s
+ */
+public interface StreamDependencyTree {
+
+    /**
+     * Get all the ancestors of the given {@link StreamId}. NOTE: this INCLUDES the source {@link StreamId}!
+     */
+    Set<StreamId<?>> getAncestorsFrom(StreamId<?> source);
```

Comment:
we went for ancestor since it makes it very clear what they are. Parent could also work, but I would not introduce "upstream" concept since it, possibly, introduces the usual problem with "direction" (what is up and down) hahah
Do you have a strong preference with parent instead of ancestor?